arquillian-ape-nosql/infinispan/src/main/java/org/arquillian/ape/nosql/infinispan/InfinispanPopulatorService.java

```diff
@@ -0,0 +1,79 @@
+package org.arquillian.ape.nosql.infinispan;
+
+import com.lordofthejars.nosqlunit.infinispan.DefaultInfinispanInsertionStrategy;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import org.arquillian.ape.core.DataSetLoader;
+import org.arquillian.ape.nosql.NoSqlPopulatorService;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.commons.api.BasicCache;
+
+class InfinispanPopulatorService implements NoSqlPopulatorService<Infinispan> {
+
+    private BasicCache<Object, Object> infinispanBasicCache;
+
+    @Override
+    public void connect(String host, int port, String database, Map<String, Object> customOptions) {
+
+        final ConfigurationBuilder configurationBuilder = new ConfigurationBuilder();
+
+        configurationBuilder.addServer()
+            .host(host)
+            .port(port);
+
+        final InfinispanOptions infinispanOptions = new InfinispanOptions(customOptions);
+        infinispanOptions.configure(configurationBuilder);
+
+        final RemoteCacheManager remoteCacheManager = new RemoteCacheManager(configurationBuilder.build());
+
+        if (database == null) {
+            infinispanBasicCache = remoteCacheManager.getCache();
+        } else {
+            infinispanBasicCache = remoteCacheManager.getCache(database);
+        }
+    }
+
+    @Override
+    public void connect(URI uri, String database, Map<String, Object> customOptions) {
+        this.connect(uri.getHost(), uri.getPort(), database, customOptions);
+    }
+
+
+    @Override
+    public void disconnect() {
+        if (infinispanBasicCache != null) {
+            infinispanBasicCache.stop();
+        }
+    }
+
+    @Override
+    public void execute(List<String> resources) {
+        final DefaultInfinispanInsertionStrategy defaultInfinispanInsertionStrategy = new DefaultInfinispanInsertionStrategy();
+
+        resources.stream()
+            .map(DataSetLoader::resolve)
+            .forEach((InputStream dataset) -> {
+                try {
+                    defaultInfinispanInsertionStrategy.insert(() -> infinispanBasicCache, dataset);
+                } catch (Throwable e) {
+                    throw new IllegalStateException(e);
```

Comment:
Maybe we can add a short message under what circumstances this error happened? Otherwise, it might not be so obvious from the stack trace.